sdk/batch/batch/test/batchClient.spec.ts

```diff
@@ -0,0 +1,1232 @@
+      );
+
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should create a task with authentication token settings successfully", async () => {
+      const jobId = "HelloWorldJobNodeSDKTest";
+      const taskId = "TaskWithAuthTokenSettings";
+      const task: BatchServiceModels.TaskAddParameter = {
+        id: taskId,
+        commandLine: "cmd /c echo Hello World",
+        authenticationTokenSettings: {
+          access: ["job"]
+        }
+      };
+
+      const result = await client.task.add(jobId, task);
+
+      assert.equal(result._response.status, 201);
+
+      const result2 = await client.task.get(jobId, taskId);
+
+      assert.isDefined(result2.authenticationTokenSettings);
+      assert.isDefined(result2.authenticationTokenSettings!.access);
+      assert.lengthOf(result2.authenticationTokenSettings!.access!, 1);
+      assert.equal(result2.authenticationTokenSettings!.access![0], "job");
+    });
+
+    it("should create a task with a user identity successfully", async () => {
+      const jobId = "HelloWorldJobNodeSDKTest";
+      const taskId = "TaskWithUserIdentity";
+      const task = {
+        id: taskId,
+        // This command should return a non-zero exit code for a non-admin user
+        commandLine: "cmd /c net session >nul 2>&1",
+        userIdentity: {
+          userName: nonAdminPoolUser
+        }
+      };
+
+      const result = await client.task.add(jobId, task);
+
+      assert.equal(result._response.status, 201);
+
+      wait(15000);
+      const result2 = await client.task.get(jobId, taskId);
+
+      assert.isDefined(result2.userIdentity);
+      assert.equal(result2.userIdentity!.userName, nonAdminPoolUser);
+      assert.isDefined(result2.executionInfo);
+      assert.equal(result2.executionInfo!.result, "failure");
+      assert.notEqual(result2.executionInfo!.exitCode, 0);
+    });
+
+    it("should count tasks sucessfully", async () => {
+      const jobId = "HelloWorldJobNodeSDKTest";
+      const result = await client.job.getTaskCounts(jobId);
+
+      assert.isDefined(result.active);
+      assert.isDefined(result.completed);
+    });
+
+    it("should list files from task successfully", async () => {
+      const result = await client.file.listFromTask(
+        "HelloWorldJobNodeSDKTest",
+        "HelloWorldNodeSDKTestTask2"
+      );
+
+      assert.isAtLeast(result.length, 1);
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should get file properties from task successfully", async () => {
+      const result = await client.file.getPropertiesFromTask(
+        "HelloWorldJobNodeSDKTest",
+        "HelloWorldNodeSDKTestTask2",
+        "stderr.txt"
+      );
+
+      assert.equal(result._response.status, 200);
+    });
+
+    it("should get file from task successfully", (done) => {
+      client.file
+        .getFromTask("HelloWorldJobNodeSDKTest", "HelloWorldNodeSDKTestTask2", "stdout.txt")
+        .then((result) => {
```

Comment:
It'd be great if we can convert tests using `(done)` and promise chains to `async` tests, too keep all of them consistent

```typescript
   const result = await client.file.getFromTask("HelloWorldJobNodeSDKTest", "HelloWorldNodeSDKTestTask2", "stdout.txt");
```